packages/clay-shared/src/Portal.tsx

```diff
@@ -41,15 +40,17 @@ export const ClayPortal: React.FunctionComponent<
 				? containerRef.current
 				: closestParent;
 
-		if (elToMountTo && portalRef.current) {
-			if (otherProps?.className) {
-				portalRef.current.classList.add(otherProps.className);
-			}
-			if (otherProps?.id) {
-				portalRef.current.id = otherProps.id;
-			}
+		if (elToMountTo) {
+			const myDiv = document.createElement('div');
+
+			ReactDOM.render(<div {...otherProps}></div> , myDiv, () => {
+				portalRef.current = myDiv.firstChild as Element;
+
+				elToMountTo.appendChild(portalRef.current);
 
-			elToMountTo.appendChild(portalRef.current);
+				setTest('re-render'); //this way we force a re-render when portalRef.current has a value,
```

Comment:
Notice that this is done in the ReactDOM.render callback, so to force a re-render of the component (otherwise the `portalRef.current` is null and the createPortal was not being called here) I had to use the states of React. Probably there is a best way to do this... I'm all hears.